app/c2/gist.py

```diff
@@ -0,0 +1,151 @@
+                                        )
+        return await self._post_gist(gist)
+
+    """ PRIVATE """
+
+    async def _post_gist(self, gist):
+        headers = {'Authorization': 'token {}'.format(self.key)}
+        async with aiohttp.ClientSession(headers=headers, connector=aiohttp.TCPConnector(verify_ssl=False)) as session:
+            return await self._post(session, 'https://api.github.com/gists', body=gist)
+
+    @staticmethod
+    def _build_gist_content(comm_type, paw, files):
+        return {
+            "description": "{}-{}".format(comm_type, paw),
+            "public": False,
+            "files": files}
+
+    async def _get_gist_content(self, urls):
+        all_content = []
+        headers = {'Authorization': 'token {}'.format(self.key)}
+        for url in urls:
+            all_content.append(await self._fetch_content(url, headers))
+        return all_content
+
+    async def _fetch_content(self, url, headers):
+        async with aiohttp.ClientSession(headers=headers, connector=aiohttp.TCPConnector(verify_ssl=False)) as session:
+            return json.loads(self.decode_bytes(await self._fetch(session, url)))
+
+    async def _get_raw_gist_urls(self, comm_type):
+        raw_urls = []
+        for gist in await self._get_gists():
+            if comm_type in gist['description']:
+                for file in gist.get('files'):
+                    raw_url = gist['files'][file]['raw_url']
+                    raw_urls.append((raw_url, gist['id']))
+        return raw_urls
+
+    async def _get_gists(self):
+        headers = {'Authorization': 'token {}'.format(self.key)}
+        async with aiohttp.ClientSession(headers=headers, connector=aiohttp.TCPConnector(verify_ssl=False)) as session:
+            return json.loads(await self._fetch(session, 'https://api.github.com/gists'))
+
+    async def _delete_gists(self, gist_ids):
+        headers = {'Authorization': 'token {}'.format(self.key)}
+        for _id in gist_ids:
+            async with aiohttp.ClientSession(headers=headers,
+                                             connector=aiohttp.TCPConnector(verify_ssl=False)) as session:
+                await self._delete(session, 'https://api.github.com/gists/{}'.format(_id))
+
+    @staticmethod
+    async def _fetch(session, url):
+        async with session.get(url) as response:
+            return await response.text()
+
+    @staticmethod
+    async def _post(session, url, body):
+        async with session.post(url, json=body) as response:
+            return await response.text()
+
+    async def _wait_for_paw(self, paw, comm_type):
+        for gist in await self._get_gists():
+            if '{}-{}'.format(comm_type, paw) == gist['description']:
+                return True
+        return False
+
+    @staticmethod
+    async def _delete(session, url):
+        async with session.delete(url) as response:
+            return await response.text('ISO-8859-1')
+
+    def valid_config(self):
```

Comment:
make private with an _

Reply:
It's not a private function (overrides a super class).  Moved and added doctoring